app/models/setting.rb

```diff
@@ -68,6 +68,14 @@ def self.[](name)
     end
   end
 
+  def self.has_value?(name)
```

Comment:
This is checking if a value has already been used. Why is this different to lines 63 to 65? It seems to me that it's a very similar operation but the original one is using the Rails cache.